### PR TITLE
feat/ view a post

### PR DESCRIPTION
**Pull request by avinash360**
## Types of changes

- [x] Add tags to auto close the corresponding issue.
- [x] I followed branch and commit naming convention.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: server/controllers/posts.js [hunk 1]
```diff
@@ -246,6 +246,7 @@ const topPosts = async (req, res) => {
     ]);
     return res.status(200).json({ success: true, message: 'Posts fetched', posts });
   } catch (error) {
+    console.log('@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@2', error);
```
Comment (Navneet-Sahota) [inline server/controllers/posts.js]: Remove console statement

File: client/src/API/index.js [hunk 2]
```diff
@@ -134,6 +134,36 @@ const fetchUserPosts = async ({
   }
 };
 
+const upvoteAPost = async (postId, userId, type) => {
+  let response;
+  try {
+    response = await axios.patch(`${axiosConfig.baseURL}/posts/${postId}/${type}`, { userId }, axiosConfig);
+  } catch (err) {
+    return alert(err.response.data.message);
```
Comment (Navneet-Sahota) [inline client/src/API/index.js]: Don't use alert here. Throw err in the catch block and catch that err in where you are calling thi function. Use alert there. So we can change alerts to modal.
Check signin/sign-up function for reference

**Comment by avinash360** [comment]
functionality video : https://www.loom.com/share/ee9b37aff94e4f849f2ee4bfb37dba4e

File: client/src/API/index.js (hunk 2 above)
Comment (avinash360) [reply client/src/API/index.js]: done

File: server/controllers/posts.js (hunk 1 above)
Comment (avinash360) [reply server/controllers/posts.js]: done

**Pull request merged**